src/test/resources/java-errors-handling.xml

```diff
@@ -0,0 +1,47 @@
+<mule xmlns="http://www.mulesoft.org/schema/mule/core"
+      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+      xmlns:java="http://www.mulesoft.org/schema/mule/java"
+      xsi:schemaLocation="http://www.mulesoft.org/schema/mule/core http://www.mulesoft.org/schema/mule/core/current/mule.xsd
+          http://www.mulesoft.org/schema/mule/java http://www.mulesoft.org/schema/mule/java/current/mule-java.xsd">
+
+    <flow name="invokeInstanceWithClass">
+        <java:invoke method="#[vars.method]" instance="#[vars.instance]">
+            <java:args>
+                #[vars.args]
+            </java:args>
+        </java:invoke>
+    </flow>
+
+    <flow name="invokeInstance">
+        <java:invoke method="#[vars.method]" instance="#[vars.instance]">
+            <java:args>
+                #[vars.args]
+            </java:args>
+        </java:invoke>
+    </flow>
+
+    <flow name="invokeStatic">
+        <java:invoke-static class="${className}" method="#[vars.method]">
+            <java:args>
+                #[vars.args]
+            </java:args>
+        </java:invoke-static>
+    </flow>
+
+    <flow name="newInstance">
+        <java:new class="#[vars.class]">
```

Comment:
this is confusing. When I first read this, my though was `How would a user get a Class object in a variable in the first place?`... rename all `class` parameters to `classname`?

Reply:
if I do this then `method` should be `methodName` and it all becomes verbose... I think that the type of the field in the IDE is enough, besides it is marked as a @ClassValue, so the user will have a lookup view for classnames